CMakeLists.txt

```diff
@@ -1,19 +1,52 @@
-cmake_minimum_required (VERSION 2.8.11)
+cmake_minimum_required (VERSION 3.5)
+
+if(DEFINED ENV{VCPKG_ROOT} AND NOT DEFINED CMAKE_TOOLCHAIN_FILE)
+  set(CMAKE_TOOLCHAIN_FILE "$ENV{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake")
+endif()
+
 project (signalrclient)
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -fPIC -L -lcpprest")
+if(NOT WIN32)
+  set(EXTRA_FLAGS "-std=c++11 -fPIC -DNO_SIGNALRCLIENT_EXPORTS")
+else()
+  set(EXTRA_FLAGS "-DSIGNALRCLIENT_EXPORTS")
+endif()
+
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${EXTRA_FLAGS}")
+set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${CMAKE_LDFLAGS}")
+
+find_package(cpprestsdk CONFIG REQUIRED)
+set(CPPREST_LIB "cpprestsdk::cpprest")
```

Comment:
I spent 24 minutes building/installing the cpprestsdk on my mac using `./submodules/vcpkg/vcpkg install cpprestsdk` (*thanks C++/Boost*), but when I run cmake, I get the following error:

```
shalter@halter73-mbp build.Release (brecon/cmake% u=)$ cmake .. -DCMAKE_TOOLCHAIN_FILE=../submodules/vcpkg/scripts/buildsystems/vcpkg.cmake -DENABLE_TESTING=1 -DCMAKE_BUILD_TYPE=Release
CMake Error at CMakeLists.txt:18 (find_package):
  Could not find a package configuration file provided by "cpprestsdk" with
  any of the following names:

    cpprestsdkConfig.cmake
    cpprestsdk-config.cmake

  Add the installation prefix of "cpprestsdk" to CMAKE_PREFIX_PATH or set
  "cpprestsdk_DIR" to a directory containing one of the above files.  If
  "cpprestsdk" provides a separate development package or SDK, be sure it has
  been installed.


-- Configuring incomplete, errors occurred!
See also "/Users/shalter/source/aspnet/SignalR-Client-Cpp/CMakeFiles/CMakeOutput.log".
```

Reply:
I had the same problem :(

Reply:
I'd like to see the vcpkg folder on either (or both) of your machines

Reply:
As in the submodules/vcpkg directory? Mine is just what's in https://github.com/Microsoft/vcpkg/tree/d1110ceb7feb527ac58f274239d435e32dca4b84 without changes.

Reply:
At a bare minimum you should have a new file `vcpkg` that's runnable on your machine. But more importantly I'm looking for the submodules/vcpkg/installed folder.